pkg/config/crd/resource.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crd
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+//resource type represents the structure of a single custom resource.
+type resource struct {
+	Kind              string `json:"kind"`
+	APIVersion        string `json:"apiVersion"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
why is ObjectMeta embedded? by not meta ObjectMeta  or
Kind / APIVersion is same as  `metav1.TypeMeta`